ghcide/src/Development/IDE/Core/ProgressReporting.hs

```diff
@@ -33,119 +34,156 @@ import           Language.LSP.Server            (ProgressAmount (..),
                                                  withProgress)
 import qualified Language.LSP.Server            as LSP
 import qualified StmContainers.Map              as STM
-import           UnliftIO                       (Async, async, cancel)
+import           UnliftIO                       (Async, MonadUnliftIO, async,
+                                                 bracket, cancel)
 
 data ProgressEvent
-    = KickStarted
-    | KickCompleted
+  = ProgressStarted
+  | ProgressCompleted
 
-data ProgressReporting  = ProgressReporting
-  { progressUpdate :: ProgressEvent -> IO ()
-  , inProgress     :: forall a. NormalizedFilePath -> Action a -> Action a
-  , progressStop   :: IO ()
+data ProgressReporting m = ProgressReporting
+  { progressUpdate :: ProgressEvent -> IO (),
```

Comment:
it's a bit surprising that these are still in `IO` rather than `m`? I think they could be, since you're requiring `MonadUnliftIO` for `m` everywhere?

Reply:
right, we let's swtich it to m